bundles/org.openhab.binding.ntp/src/main/java/org/openhab/binding/ntp/internal/handler/NtpHandler.java

```diff
@@ -54,217 +54,183 @@
     private static final int NTP_TIMEOUT = 30000;
 
     public static final String DATE_PATTERN_WITH_TZ = "yyyy-MM-dd HH:mm:ss z";
+    private static final DateTimeFormatter DATE_FORMATTER_WITH_TZ = DateTimeFormatter.ofPattern(DATE_PATTERN_WITH_TZ);
 
-    /** for logging purposes */
-    private final DateFormat SDF = new SimpleDateFormat(DATE_PATTERN_WITH_TZ);
+    private final Logger logger = LoggerFactory.getLogger(NtpHandler.class);
 
     /** for publish purposes */
     private DateTimeFormatter dateTimeFormat = DateTimeFormatter.ofPattern(DATE_PATTERN_WITH_TZ);
 
-    private final LocaleProvider localeProvider;
+    private @Nullable NtpThingConfiguration configuration;
 
-    ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> refreshJob;
 
-    /** NTP host */
-    private String hostname;
-    /** NTP server port */
-    private BigDecimal port;
-    /** refresh interval */
-    private BigDecimal refreshInterval;
-    /** NTP refresh frequency */
-    private BigDecimal refreshNtp = new BigDecimal(0);
-    /** Timezone */
-    private TimeZone timeZone;
-    /** Locale */
-    private Locale locale;
+    private ZoneId defaultTimeZoneId;
+    private ZoneId timeZoneId;
 
     /** NTP refresh counter */
     private int refreshNtpCount = 0;
     /** NTP system time delta */
     private long timeOffset;
 
-    private ChannelUID dateTimeChannelUID;
-    private ChannelUID stringChannelUID;
-
-    public NtpHandler(final Thing thing, final LocaleProvider localeProvider) {
+    public NtpHandler(final Thing thing, final TimeZoneProvider timeZoneProvider) {
         super(thing);
-        this.localeProvider = localeProvider;
+        this.defaultTimeZoneId = timeZoneProvider.getTimeZone();
+        this.timeZoneId = defaultTimeZoneId;
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        // No specific commands tied to this, but we will trigger an update
-        this.refreshNtpCount = 0;
-        refreshTimeDate();
+        if (command == RefreshType.REFRESH) {
+            logger.debug("Refreshing channel '{}' for '{}'.", channelUID.getId(), getThing().getUID());
+            refreshTimeDate();
+        }
     }
 
     @Override
     public void initialize() {
-        try {
-            logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
+        logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
 
-            Configuration config = getThing().getConfiguration();
-            hostname = config.get(PROPERTY_NTP_SERVER_HOST).toString();
-            port = (BigDecimal) config.get(PROPERTY_NTP_SERVER_PORT);
-            refreshInterval = (BigDecimal) config.get(PROPERTY_REFRESH_INTERVAL);
-            refreshNtp = (BigDecimal) config.get(PROPERTY_REFRESH_NTP);
-            refreshNtpCount = 0;
+        NtpThingConfiguration config = getConfigAs(NtpThingConfiguration.class);
+        configuration = config;
```

Comment:
Why not use
```suggestion
        configuration = getConfigAs(NtpThingConfiguration.class);
```

Reply:
I was to avoid null check warnings due to configuration which was nullable.
Now that configuration is non null, the local variable can be avoided.

Reply:
Local variable removed.